src/NServiceBus.Testing/ExpectationException.cs

```diff
@@ -0,0 +1,24 @@
+﻿namespace NServiceBus.Testing
+{
+    using System;
+
+    /// <summary>
+    /// Exception representing a failed expectation.
+    /// </summary>
+    public class ExpectationException : Exception
```

Comment:
It allows users to use `Assert.Throws<ExpectationException>` to not hide other `Exception` thrown by the code path.